Source/Particles/LaserParticleContainer.cpp

```diff
@@ -477,7 +477,7 @@ LaserParticleContainer::Evolve (int lev,
             plane_Yp.resize(np);
             amplitude_E.resize(np);
 
-            if (rho) {
+            if (rho && ! skip_deposition) {
```

Comment:
Will your multi-J scheme also affect the laser deposition? If not, this is probably not needed and instead you might have an assert that the skip_deposition must be false. I don't think it is checked anywhere if a laser is used when using electrostatic.

Reply:
That's a good point. At this point, I think it would still make sense to apply the multi-J scheme to the laser deposition too. We could later experiment with adding/removing it.